### PR TITLE
fix(es/typescript): transform namespace with multiple identifiers in name

### LINKED ISSUE #2810: Namespace with multiple identifiers in name does not end up in output
### Describe the bug

Create a namespace with multiple identifiers (technically nested namespaces) and it will not end up in the emitted output.

### Input code

```typescript
namespace Test.Test {
  export const Value = 1;
}

console.log(Test.Test.Value);
```

### Config

_No response_

### Playground link

https://play.swc.rs/?version=1.2.111&code=H4sIAAAAAAAAA8tLzE0tLkhMTlUISS0u0QMRCtW8XAoKqRUF%2BUUlCsn5eUCRsMSc0lQFWwVDa16uWl4uXi6QcH5Oql5OfroGXKMeWJmmNQDjeL5KVAAAAA%3D%3D&config=H4sIAAAAAAAAA3VTy47bMAz8F533sFugxWI%2FoLd%2Bg6FYlMOtLQok5U0Q5N9LxY4fyfYmjmakEYe6uE9p3cfFZc8CXFdyTupP7sPpOYO0jFndi1MxSLnAiwvQEnsllgm52q7nDtQ0ID9e334avycSuCsGTBjP9fCWhswgUtemKQMktSL6XoznmelL7qoDUQ8%2BPdWNlwaTQge8Slvqe58FmtHzoqi3eUahtIOKQmgyU97KU0BFSlYuYAAfmpYCrAgytIojbEh2jpGSmLk9GOBQuq42dboaRt8Xr0tX4HRrhV266I6Eok0sSR6g2e0knKDpoTMLY8OghdOd80mYdr34C2A%2Bey%2BS%2FLDxf8OjZbFnxm9YmKIFqecFtZBXVwk6e1yDGJcXVtfAirCQGEJpob6wXQ%2Bfwd2DBAM0EKO1exHLF2p73LiqE0px2be2%2BbiGMJXNfaj2aJ2DJ%2FC32dJtHjM%2BeD0%2BYnIeDtQ%2FHTGAHik8wdYJpUeQbYRP%2BREtKYD1GcJmo0gt5882%2BNTVWbs4pdzDCP0%2Bpv%2BG%2FA2WGcea2v43ILyvQXgbDnx7nYHrtTqgUGYHlsDt2%2F9y1%2FWfT3Gg%2FJl5k%2FIfkYuaqmwEAAA%3D

### Expected behavior

Should not remove the namespace.

### Actual behaviour

Outputs:

```js
console.log(Test.Test.Value);
```

### Version

1.2.111

### Additional context

See here:

https://github.com/swc-project/swc/blob/372f5bf1e0608d8a4ed7fe4376839b1c0ffeed27/crates/swc_ecma_transforms_typescript/src/strip.rs#L1093

### PR DESCRIPTION
**Pull request by dsherret**
~~I'm going to wait until #2940 and #2951 are merged before running the tests on this one (don't want to deal with conflicts).~~

~~This includes #2951 for now to make it easier for me to develop because I'm going to open another PR on top of this one.~~

Closes #2810

### CODE REVIEW

File: crates/swc_ecma_transforms_typescript/src/strip.rs [hunk 1]
```diff
@@ -1083,32 +1083,172 @@ where
         if module.global || module.declare {
             return None;
         }
-        let module_span = module.span;
```
Comment (dsherret) [inline crates/swc_ecma_transforms_typescript/src/strip.rs:1086]: I will add this back and other spans later. I see it's necessary for comments.

File: crates/swc_ecma_transforms_typescript/src/strip.rs [hunk 2]
```diff
@@ -1086,32 +1086,186 @@ where
...
+    fn handle_ts_namespace_decl(
+        &mut self,
+        decl: TsNamespaceDecl,
+        parent_private_name: &Ident,
+    ) -> Option<Vec<Stmt>> {
+        let private_name = private_ident!(decl.id.sym.clone());
+        // we'll always create a variable in this scenario
+        let var_id = Ident::new(
+            decl.id.sym.clone(),
+            DUMMY_SP.with_ctxt(decl.id.span.ctxt.clone()),
```
Comment (kdy1) [inline crates/swc_ecma_transforms_typescript/src/strip.rs:1131]: This `.clone()` seems redundant as `SyntaxContext` is `Copy`.

**Review by kdy1 — CHANGES_REQUESTED** [review]
It's mostly fine, thanks!
I found one code style issue.

File: crates/swc_ecma_transforms_typescript/src/strip.rs (hunk 2 above)
Comment (dsherret) [reply crates/swc_ecma_transforms_typescript/src/strip.rs]: Done in 794961694a05c77d529c106ece5ec82a936c096d

**Review by kdy1 — APPROVED** [review]
Thanks!

**Pull request merged**